src/ecrc-frontend/src/components/page/bcscRedirect/BcscRedirect.test.js

```diff
@@ -1,24 +1,93 @@
 import React from "react";
-import { create } from "react-test-renderer";
+import axios from "axios";
+import { MemoryRouter, Router } from "react-router-dom";
+import { render, fireEvent, getByText, wait } from "@testing-library/react";
+import { createMemoryHistory } from "history";
+import { generateJWTToken } from "../../../modules/AuthenticationHelper";
 
 import BcscRedirect from "./BcscRedirect";
 
+jest.mock("axios");
+
 describe("BcscRedirect Page Component", () => {
-  test("Matches the snapshot", () => {
-    const header = {
-      name: "Criminal Record Check"
-    };
-
-    const saveOrg = () => jest.fn();
-    const setError = () => jest.fn();
-
-    const page = {
-      header,
-      saveOrg,
-      setError
-    };
-
-    const bcscRedirect = create(<BcscRedirect page={page} />);
-    expect(bcscRedirect.toJSON()).toMatchSnapshot();
+  window.scrollTo = jest.fn();
+  window.open = jest.fn().mockImplementation(() => true);
+
+  const header = {
+    name: "Criminal Record Check"
+  };
+
+  const saveOrg = jest.fn();
+  const setError = jest.fn();
+
+  const page = {
+    header,
+    saveOrg,
+    setError
+  };
+
+  const axiosCall = {
+    data: "http://test.com"
+  };
+
+  beforeEach(() => {
+    sessionStorage.setItem("validator", "secret");
+    sessionStorage.setItem("uuid", "unique123");
+    generateJWTToken({
+      actionsPerformed: ["tou"],
+      authorities: ["Authorized"]
+    });
+  });
+
+  test("Matches the snapshot", async () => {
+    axios.get.mockResolvedValueOnce(axiosCall);
+    const { asFragment } = render(
+      <MemoryRouter>
+        <BcscRedirect page={page} />
+      </MemoryRouter>
+    );
+    await wait(() => {});
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test("Validate Login button", async () => {
+    axios.get.mockResolvedValueOnce(axiosCall);
+    const { container } = render(
+      <MemoryRouter>
+        <BcscRedirect page={page} />
+      </MemoryRouter>
+    );
+    await wait(() => {});
+    fireEvent.click(getByText(container, "Login with a BC Services Card"));
+    expect(saveOrg).toHaveBeenCalled();
+    expect(window.open).toHaveBeenCalled();
+  });
+
+  test("Validate Redirect to Home when unauthorized", async () => {
```

Comment:
based on the jwt you are generating, this test isnt actually for unauthorized, it is for inadequate actions performed

Reply:
The test is for an if condition that checks for authentication and for actions performed. I think we could leave it as unauthorized as a blanket to cover unauthenticated and not allowed to perform a function as it all satisfies the same condition clause.